src/tests/test_pipeline.cpp

```diff
@@ -356,3 +357,100 @@ TEST_F(TestPipeline, RequiresMoreManifolds)
 
     EXPECT_ANY_THROW(exec1.register_pipeline(std::move(pipe)));
 }
+
+class Buffer
+{
+  public:
+    Buffer() = default;
+
+    DELETE_COPYABILITY(Buffer);
+
+    std::size_t* data()
+    {
+        return m_buffer.data();
+    }
+
+    const std::size_t* data() const
+    {
+        return m_buffer.data();
+    }
+
+    std::size_t size() const
+    {
+        return m_buffer.size();
+    }
+
+  private:
+    std::array<std::size_t, 1024> m_buffer;
+};
+
+TEST_F(TestPipeline, ReusablePool)
+{
+    auto pool = data::ReusablePool<Buffer>::create(32);
```

Comment:
Would like to see additional unit tests, at least one for each new object, and which exercise the full API so its clear its working as expected.